hooks/pre_gen_project.py

```diff
@@ -0,0 +1,11 @@
+import re
+import sys
+
+MODULE_REGEX = r'^[_a-zA-Z][_a-zA-Z0-9]+$'
```

Comment:
I'm not sure this regex is accurate. Python packages should only use alphanumeric characters, not start with a number and contain only hyphens, not underscores.

Check:
- https://peps.python.org/pep-0008/#package-and-module-names
- https://peps.python.org/pep-0423/

However, it's a little weird (check discussion on Stack Overflow etc):
- Distribution names should contain hyphens, but not underscores
- Import names should contain underscores, but not hyphens

So basically, say we want to create a tool called "My Tool".

We could create a slug `my-tool` and use it as a distribution name.
However, the _code_ should be in a folder called `my_tool`.

I think we should try to enforce this and convert white spaces and underscores in project names accordingly to create slugs - and use them correctly for each purpose (distribution name, code directory/import name).

Reply:
regex is taken from the [docs](https://cookiecutter.readthedocs.io/en/1.7.3/advanced/hooks.html?highlight=pre%20gen%20name#example-validating-template-variables), added project_name field and that could by `AnY_-Th!ng` but th `project_slug` will be the name of package that will be validated.

Reply:
Okay, fine - but would be good to create a default project slug (for people to accept when initializing the project via the cookiecutter) from the project name according to some simple rules.